releasing/local-build.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+#
+# Works exactly like cloudbuild.sh but doesn't perform a release.
+#
+# Usage (from top of repo):
+#
+#  releasing/local-build.sh TAG [--snapshot]
```

Comment:
nit: Can we name this file `localbuild.sh`?

Reply:
I tried to @natasha41575 (I had two explicit steps to facilitate a rename to prevent conflicts with the original file). However, it originally looked like the CI bot suggested I shouldn't have multiple commits https://github.com/kubernetes-sigs/kustomize/pull/4262#issuecomment-954412269. So to compact this into a single commit while _also_ ensuring it is tracked as a rename (to preserve commit history), I just chose to name this new file `local-build.sh` instead. That way Git would allow for a rename of the original old `localbuild.sh` -> `cloudbuild-local.sh`.

I'm guessing it's fine to have multiple commits? If so, it should be possible to easily reuse the original `localbuild.sh` and reduce the overall merge diff size.

Reply:
Don't worry about that bot--it's just trying to be helpful by letting you know that if you _want_ to rebase into a single commit before merging, it can do that for you. Multiple commits to preserve the rename makes sense to me.

Reply:
Alright great, thanks! Good to know it's not entirely faux paus. I also personally prefer the extra commit explaining why it's being done, especially when things shift around like this.